armi/reactor/blocks.py

```diff
@@ -1580,61 +1580,85 @@ def setPinPowers(
         numPins,
         imax,
         jmax,
-        gamma=False,
         removeSixCornerPins=False,
         powerKeySuffix="",
     ):
         """
-        Updates the pin powers of this block for the current rotation.
+        Updates the pin linear power densities of this block for the current rotation.
+        The linear densities are represented by the *linPowByPin* parameter.
 
         Parameters
         ----------
-        powers : list of floats
-            The block-level pin linear power densities. pinPowers[i] represents the average linear
-            power density of pin i.
-            Power units are Watts/cm (Watts produced per cm of pin length).
-            The "ARMI pin ordering" is used, which is counter-clockwise from 3 o'clock.
+        powers : list of floats, required
+            The block-level pin linear power densities. powers[i] represents the average
+            linear power density of pin i. The units of linear power density is watts/cm
+            (i.e., watts produced per cm of pin length). The "ARMI pin ordering" must be
+            be used, which is counter-clockwise from 3 o'clock.
 
-        Notes
-        -----
-        This handles rotations using the pinLocation parameters.
+        numPins : int, required
+            Number of pins in the block. This parameter should probably be removed because it
+            can be derived from the block.
 
-        This sets:
+        imax: int, required
+            Number of pin rings. This parameter should probably be removed because it
+            can be derived from the block.
 
-        self.p.pinPowers : 1-D numpy array
-            The block-level pin linear power densities. pinPowers[i] represents the average linear
-            power density of pin i.
-            Power units are Watts/cm (Watts produced per cm of pin length).
-            The "ARMI pin ordering" is used, which is counter-clockwise from 3 o'clock.
+        jmax: list of ints, required
+            Number of pins per ring. Element 0 is the number of pins in ring 1, element 1
+            is the number of pins in ring 2, and so on. This parameter should probably be
+            removed because it can be derived from the block.
+
+        removeSixCornerPins: bool, optional.
+            Defaults to False
```

Comment:
This is silly. We should just have the block design describe where the pins are. This should be removed.